src/sdl2_extras.c

```diff
@@ -0,0 +1,198 @@
+//
+// sdl2_extras.c
+// sdl2-sparkling
+//
+// Created by Antonio Sarmento
+// on 03/04/2016
+//
+// Licensed under the 2-clause BSD License
+//
+
+#include "sdl2_extras.h"
+#include "sdl2_sparkling.h"
+
+/////////////////////////////////
+//////   Just some paths   //////
+/////////////////////////////////
+static const char *get_base_path(void)
+{
+	const char *base = SDL_GetBasePath();
+	return base ? base : "";
+}
+
+static const char *get_pref_path(const char *org, const char *app)
+{
+	const char *pref = NULL;
+
+	if (org != NULL && app != NULL) {
+		pref = SDL_GetPrefPath(org, app);
+	}
+
+	return pref ? pref : "";
+}
+
+int spnlib_SDL_GetPaths(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	if (argc >= 2) {
+		CHECK_ARG_RETURN_ON_ERROR(0, string);
+		CHECK_ARG_RETURN_ON_ERROR(1, string);
+	}
+
+	// construct return value: hashmap with strings
+	*ret = spn_makehashmap();
+	SpnHashMap *paths = spn_hashmapvalue(ret);
+
+	// get proper arguments for Pref, if given
+	const char *org = NULL, *app = NULL;
+	if (argc >= 2) {
+		org = STRARG(0);
+		app = STRARG(1);
+	}
+
+	SpnValue base = spn_makestring_nocopy(get_base_path());
+	SpnValue pref = spn_makestring_nocopy(get_pref_path(org, app));
+
+	spn_hashmap_set_strkey(paths, "base", &base);
+	spn_hashmap_set_strkey(paths, "pref", &pref);
+
+	spn_value_release(&base);
+	spn_value_release(&pref);
+
+	return 0;
+}
+
+/////////////////////////////////
+//////    SDL's version    //////
+/////////////////////////////////
+int spnlib_SDL_GetVersion(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	*ret = spn_makehashmap();
+	SpnHashMap *version = spn_hashmapvalue(ret);
+
+	SDL_version linked;
+	SDL_GetVersion(&linked);
+
+	SpnValue major = spn_makeint(linked.major);
+	SpnValue minor = spn_makeint(linked.minor);
+	SpnValue patch = spn_makeint(linked.patch);
+
+	spn_hashmap_set_strkey(version, "major", &major);
+	spn_hashmap_set_strkey(version, "minor", &minor);
+	spn_hashmap_set_strkey(version, "patch", &patch);
+
+	spn_value_release(&major);
+	spn_value_release(&minor);
+	spn_value_release(&patch);
+
+	return 0;
+}
+
+/////////////////////////////////
+////    Platform and CPU    /////
+/////////////////////////////////
+int spnlib_SDL_GetPlatform(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	// Simplest function to bind
+	*ret = spn_makestring_nocopy(SDL_GetPlatform());
+
```

Comment:
there's an unnecessary empty line between setting the Sparkling return value and the return from the C function.